misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/Main.java

```diff
@@ -18,63 +18,119 @@
 package org.ballerinalang.test.runtime;
 
 import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
 import io.ballerina.runtime.internal.launch.LaunchUtils;
 import org.ballerinalang.test.runtime.entity.ModuleStatus;
+import org.ballerinalang.test.runtime.entity.TestReport;
 import org.ballerinalang.test.runtime.entity.TestSuite;
 import org.ballerinalang.test.runtime.util.TesterinaConstants;
 import org.ballerinalang.test.runtime.util.TesterinaUtils;
+import org.wso2.ballerinalang.util.Lists;
 
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.OutputStreamWriter;
+import java.io.PrintStream;
 import java.io.Writer;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
 
 /**
  * Main class to init the test suit.
  */
 public class Main {
+    private static final PrintStream out = System.out;
+    static TestReport testReport;
+    static ClassLoader classLoader;
+
     public static void main(String[] args) throws IOException {
-        Path jsonCachePath = Paths.get(args[0]).resolve(TesterinaConstants.TESTERINA_TEST_SUITE);
-        Path jsonTmpSummaryPath = Paths.get(args[0], TesterinaConstants.STATUS_FILE);
-        String[] configArgs = Arrays.copyOfRange(args, 1, args.length);
-        LaunchUtils.initConfigurations(configArgs);
+        int exitStatus = 0;
+        int result;
 
-        try (BufferedReader br = Files.newBufferedReader(jsonCachePath, StandardCharsets.UTF_8)) {
-            //convert the json string back to object
-            Gson gson = new Gson();
-            TestSuite response = gson.fromJson(br, TestSuite.class);
-            response.setModuleName(filterModuleName(args[args.length - 1]));
-            startTestSuit(Paths.get(response.getSourceRootPath()), response, jsonTmpSummaryPath);
+        Path testCache = Paths.get(args[0]);
+        String target = args[1];
+        boolean report = Boolean.valueOf(args[2]);
+        boolean coverage = Boolean.valueOf(args[3]);
```

Comment:
Shall we add an array length check before accessing these array elements as well?

Reply:
Added array length check